dds/DCPS/Message_Block_Ptr.h

```diff
@@ -17,7 +17,12 @@ namespace DCPS {
 struct Message_Block_Deleter
 {
   void operator()(ACE_Message_Block* ptr) const {
-    ACE_Message_Block::release(ptr);
+    while (ptr) {
```

Comment:
```suggestion
    // In order to avoid locking order issues for message blocks with different locking strategies,
    // it is safer to unlink elements in the chain before releasing them individually
    while (ptr) {
```